docs/pages/push-notifications/sending-notifications-custom.md

```diff
@@ -0,0 +1,174 @@
+---
+title: Sending Notifications with APNS & FCM
+---
+
+Communicating directly with APNS and FCM is much more complicated than sending notifications through [Expo's push notification service](../sending-notifications/), so you should only use this feature if you're prepared to undertake that complexity. Here are a few things you'll have to handle yourself if you choose to write your own server for FCM and APNS:
+
+- Differentiating between native iOS & Android device tokens on your backend
+- Twice the amount of backend code to write and maintain (code for communicating with FCM, and then code for communicating with APNS)
+- Fetching responses from FCM and APNS to check if your notification went through, error handling, credentials management
+
+That being said, sometimes you need finer-grained control over your notifications, so communicating directly with APNS and FCM is necessary. Expo will never lock you in and force you to use our services, so the `expo-notifications` API is push service agnostic (you can use it with any push notification service).
+
+## How do I write my own APNS & FCM servers?
+
+Before we begin communicating directly with APNS & FCM, there is one client-side change you'll need to make in your app. When using Expo's notification service, you collect the `ExponentPushToken` with [`getExpoPushTokenAsync`](../../versions/latest/sdk/notifications/#getexpopushtokenasyncoptions-expotokenoptions-expopushtoken). Now that you're bypassing Expo, you'll need to collect the native device token instead with [`getDevicePushTokenAsync`](../../versions/latest/sdk/notifications/#getdevicepushtokenasync-devicepushtoken).
+
+```diff
+import * as Notifications from 'expo-notifications';
+...
+- const token = (await Notifications.getExpoPushTokenAsync()).data;
++ const token = (await Notifications.getDevicePushTokenAsync()).data:
+// send token off to your server
+```
+
+Now that you have your native device token, we can start to implement our servers:
+
+## FCM Server
+
+> This documentation is based off of [Google's documentation](https://firebase.google.com/docs/cloud-messaging/http-server-ref), and we're just going to cover the basics here to get you started.
+
+Communicating with FCM is as simple as sending a POST request, but before sending or receiving any notifications, you'll need to follow the steps [in this documentation](../using-fcm/) to configure FCM (and get your `FCM-SERVER-KEY`).
+
+```js
+await fetch('https://fcm.googleapis.com/fcm/send', {
+  method: 'POST',
+  headers: {
+    'Content-Type': 'application/json',
+    Authorization: `key=<FCM-SERVER-KEY>`,
+  },
+  body: JSON.stringify({
+    to: '<NATIVE-DEVICE-PUSH-TOKEN>',
+    priority: 'normal',
+    data: {
+      experienceId: '@yourExpoUsername/yourProjectSlug',
+      title: "\uD83D\uDCE7 You've got mail",
+      message: 'Hello world! \uD83C\uDF10',
+    },
+  }),
+});
```

Comment:
This is their legacy API, the v1 API looks different. I think they got rid of FCM API keys and instead do a credentials dance with a google application credentials JSON file.

We don't use the v1 API right now but are laying the groundwork if we ever decided we needed to and it made sense to spend time on it.

Reply:
Yeah, but think it's better in the case of this over-simplistic example to use the same credentials that a reader would already have from using the Expo notification service, rather than walk them through creating new credentials here, as well. We can add a note referencing the migration guide for the FCM v1 API though

Reply:
Maybe just mention then that this is for FCM's legacy API.

Reply:
added - https://github.com/expo/expo/pull/9935/files#diff-987d102b1b6c85d2c4dd00cf253dd3f2R33